test/Microsoft.ML.Core.Tests/UnitTests/TestUtilities.cs

```diff
@@ -0,0 +1,249 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.Trainers.FastTree.Internal;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Core.Tests.UnitTests
+{
+    public class TestUtilities : BaseTestBaseline
+    {
+        public TestUtilities(ITestOutputHelper helper)
+            : base(helper)
+        {
+        }
+
+        [Fact]
+        [TestCategory("Utilities")]
+        public void CheckIsSortedInt()
+        {
+            // A sorted (increasing) array
+            int[] x = Enumerable.Range(0, 10).ToArray();
+            Assert.True(Utils.IsSorted(x));
+
+            // A monotonically increasing array
+            var x1Temp = x[1];
+            var x7Temp = x[7];
+            x[1] = x[0];
+            x[7] = x[6];
+            Assert.True(Utils.IsSorted(x));
+            x[1] = x1Temp;
+            x[7] = x7Temp;
+
+            // Not sorted
+            x[1] = x[6];
+            Assert.False(Utils.IsSorted(x));
+            x[1] = x1Temp;
+        }
+
+        [Fact]
+        [TestCategory("Utilities")]
+        public void CheckIsSortedFloat()
+        {
+            // A sorted (increasing) array
+            List<float> x = Enumerable.Range(0, 1000000).Select(i => (float)i).ToList();
+            Assert.True(Utils.IsSorted(x));
+
+            // A monotonically increasing array
+            var x1Temp = x[1];
+            var x7Temp = x[7];
+            x[1] = x[0];
+            x[7] = x[6];
+            Assert.True(Utils.IsSorted(x));
+            x[1] = x1Temp;
+            x[7] = x7Temp;
+
+            // Not sorted
+            x[1] = x[6];
+            Assert.False(Utils.IsSorted(x));
+            x[1] = x1Temp;
+            
+            // NaN: `Array.Sort()` will put NaNs into the first position,
+            // but we want to guarantee that NaNs aren't allowed in these arrays.
+            var x0Temp = x[0];
+            x[0] = float.NaN;
+            Assert.False(Utils.IsSorted(x));
+            x[0] = x0Temp;
+        }
+
+        [Fact]
+        [TestCategory("Utilities")]
+        public void CheckIsIncreasing()
```

Comment:
>CheckIsIncreasing [](start = 20, length = 17)

Same here. Could you check the behavior with null input?

Reply:
Done!